packages/cozy-doctypes/src/Folder.js

```diff
@@ -0,0 +1,54 @@
+const Application = require('./Application')
+const CozyFile = require('./File')
+
+/**
+ * Class representing the folder model.
+ * @extends CozyFile
+ */
+class CozyFolder extends CozyFile {
+  static async getReferencedFolders(client, reference) {
+    const { included } = await client
+      .collection(CozyFile.doctype)
+      .findReferencedBy({
+        _type: Application.doctype,
+        _id: reference
+      })
+    return included.filter(
+      folder => !/^\/\.cozy_trash/.test(folder.attributes.path)
+    )
+  }
+
+  static async getOrCreateFolderWithReference(client, path, reference) {
+    const existingFolders = await CozyFolder.getReferencedFolders(
+      client,
+      reference
+    )
+
+    if (existingFolders.length) return existingFolders[0]
+
+    const collection = client.collection(CozyFile.doctype)
+    const dirId = await collection.ensureDirectoryExists(path)
+    await collection.addReferencesTo(
+      {
+        _id: reference,
+        _type: Application.doctype
+      },
+      [
+        {
+          _id: dirId
+        }
+      ]
+    )
+
+    const { data: dirInfos } = await collection.get(dirId)
+
+    return dirInfos
+  }
+}
+
+CozyFolder.REF_ADMINISTRATIVE = `${Application.doctype}/administrative`
```

Comment:
I suppose those are references kept in a special attribute of the folder, enabling an application to find a specific folder even if it has moved ? Can you add a comment ?

Instead of adding the refs directly at the root of the class, they could be namespaced under `CozyFolder.refs`.

`CozyFolder.refs.PHOTOS`

Reply:
Love that, thanks !